ros_buildfarm/templates/doc/doc_independent_docker_job.xml.em

```diff
@@ -81,14 +87,18 @@
 @{
 import os
 doc_repository_name = os.path.splitext(os.path.basename(doc_repository_url))[0]
+
+if doc_repository_branch is None:
+  doc_repository_branch = ''
+else:
+  doc_repository_branch = '--no-single-branch -b ' + doc_repository_branch
```

Comment:
For the current work on ros2_documentation, it uses the `sphinx-multiversion` project.  It works by taking a list of branches from `conf.py`, checking out each of the branches in turn and running sphinx against each one.  So we need *all* of the branches available for this to work.  This is also why I removed the `rm -fr .git` below.